src/kOS/Function/Misc.cs

```diff
@@ -18,6 +18,51 @@ public override void Execute(SharedObjects shared)
         }
     }
 
+	[Function("hudtxt")]
+	public class FunctionHudTxt : FunctionBase
+	{
+
+		public override void Execute(SharedObjects shared)
+
+		{
+			int mirror 			= Convert.ToInt32(shared.Cpu.PopValue());
+			string color 		= shared.Cpu.PopValue().ToString();
+			int size 			= Convert.ToInt32(shared.Cpu.PopValue());	
+			int style 			= Convert.ToInt32(shared.Cpu.PopValue());
+			int delay 			= Convert.ToInt32(shared.Cpu.PopValue());	
+			string textToHud 	= shared.Cpu.PopValue().ToString();
+
+			if (style == 1)
+				{
+				ScreenMessages.PostScreenMessage("<color="+color+"><size="+size+">"+textToHud+"</size></color>", delay, ScreenMessageStyle.UPPER_LEFT);
+				}
+			else if (style == 2)
+				{
+				ScreenMessages.PostScreenMessage("<color="+color+"><size="+size+">"+textToHud+"</size></color>", delay, ScreenMessageStyle.UPPER_CENTER);
+				}
+			else if (style == 3)
+				{
+				ScreenMessages.PostScreenMessage("<color="+color+"><size="+size+">"+textToHud+"</size></color>", delay, ScreenMessageStyle.UPPER_RIGHT);
+				}
+			else if (style == 4)
+				{
+				ScreenMessages.PostScreenMessage("<color="+color+"><size="+size+">"+textToHud+"</size></color>", delay, ScreenMessageStyle.LOWER_CENTER);
+				}
+			else 
+				{
+				ScreenMessages.PostScreenMessage("*"+textToHud, 3f, ScreenMessageStyle.UPPER_CENTER);
+				}
```

Comment:
Was the best of two ways. Originally I planned to convert the input as in named enum, but it would still be a decision tree. 
Looked into switch/case as well, but opinions were it would add overhead.